lib/accesslists/requirment.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package accesslists
+
+import (
+	"context"
+	"slices"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/api/types/trait"
+)
+
+// UserMeetsRequirements is a helper which will return whether the User meets the AccessList Ownership/MembershipRequires.
+func UserMeetsRequirements(identity types.User, requires accesslist.Requires) bool {
+	return newRequirementsEvaluator(identity).meets(requires)
+}
+
+// GetInheritedMembershipRequires returns the combined Requires for an Access List's members,
+// inherited from any ancestor lists, and the Access List's own MembershipRequires.
+func GetInheritedMembershipRequires(ctx context.Context, accessList *accesslist.AccessList, g AccessListAndMembersGetter) (*accesslist.Requires, error) {
```

Comment:
I understand this is just a refactor with no logical changes but looks to me a good chance to add some tests. wdyt?